backend/backend/events.py

```diff
@@ -600,25 +600,16 @@ def generate_player_jailed_event(
     <BLANKLINE>
 
     """
-    # Send the event name to the client.
-    output_stream.write('event: playerJailed\n')
-
-    # Send the JSON object which contains the elements that are not in common
-    # with the two dictionaries.
-    output_stream.write('data: ')
-
-    if not jailed_players:
-        output_stream.write(json.dumps([
-            [uid, state]
-            for uid, state in new_jailed_players.items()]))
-    else:
-        output_stream.write(json.dumps([
-            [uid, state]
-            for uid, state in new_jailed_players.items()
-            if state != jailed_players[uid]]))
+    data = []
+    for uid, state in new_jailed_players.items():
+        if not jailed_players:
+            data.append([uid, state])
+        else:
+            old_state = jailed_players[uid]
```

Comment:
I think here we’ll get a `KeyError` if `uid`is in `new_jailed_players` but not `jailed_players` still, won’t we?